docs/how-it-works.md

```diff
@@ -0,0 +1,205 @@
+
+#### Gavel's Expectations
+
+- All JSON keys on any level given in the sample must be present in the response's JSON.
+- Response's JSON values must be of the same JSON primitive type.
+- All JSON values can differ.
+- Arrays can have additional items, type or structure of the items is not validated.
+- Plain text must match perfectly.
+
+### Custom Expectations
+
+You can make your own custom expectations in [hooks](hooks.md). For instance, check out how to employ [Chai.js assertions](hooks.md#using-chai-assertions).
+
+## Making Your API Description Ready for Testing
+
+It's very likely that your API description document will not be testable __as is__. This section should help you to learn how to solve the most common issues.
+
+### URI Parameters
+
+Both [API Blueprint][] and [Swagger][] allow usage of URI templates (API Blueprint fully implements [RFC6570][], Swagger templates are much simpler). In order to have an API description which is testable, you need to describe all required parameters used in URI (path or query) and provide sample values to make Dredd able to expand URI templates with given sample values. Following rules apply when Dredd interpolates variables in a templated URI, ordered by precedence:
+
+1. Sample value (not available in Swagger).
+2. Value of `default` (only if the parameter is marked as `optional`).
+3. First value from `enum`.
+
+If Dredd isn't able to infer any value for a required parameter, it will terminate the test run and complain that the parameter is _ambiguous_.
+
+> **Note:** The implementation of API Blueprint's request-specific parameters is still in progress and there's only experimental support for it in Dredd as of now.
+
+### Request Headers
+
+In [Swagger][] documents, HTTP headers are inferred from [`"in": "header"` parameters][parameters]. HTTP headers significant for content negotiation are inferred according to following rules:
+
+- [`consumes`][] is propagated as request's `Content-Type` header.
+- [`produces`][] is propagated as request's `Accept` header.
+
+### Request Body
+
+#### API Blueprint
+
+The effective request body is taken from following places (the order goes from the highest priority to the lowest):
+
+1. [`+ Body`][] section with sample JSON payload.
+2. [`+ Attributes`][] section with data structure description in [MSON][] - API Blueprint parser automatically generates sample JSON payload from MSON.
+
+This order [exactly follows the API Blueprint specification][body-schema-attributes].
+
+#### Swagger
+
+The effective request body is inferred from [`"in": "body"` and `"in": "form"` parameters][parameters]. Since Swagger doesn't provide any way to specify raw JSON sample for the request body, Dredd's [Swagger Adapter][] generates some sample values from the JSON Schema provided in the [`schema`][] property. Following rules apply when the adapter fills values of the properties, ordered by precedence:
+
+1. Value of `default`.
```

Comment:
What about [schema examples](http://swagger.io/specification/#schemaExample)? Please verify how it works and clarify it here.